internal/connect/command_offset_describe.go

```diff
@@ -0,0 +1,180 @@
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connectorIdToName, err := c.mapConnectorIdToName(environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	offsets, err := c.V2Client.GetConnectorOffset(connectorIdToName[args[0]], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	if output.GetFormat(cmd) == output.Human {
+		return printHumanDescribeOffset(cmd, offsets, args[0], connectorIdToName[args[0]])
+	}
+
+	return printSerializedDescribeOffsets(cmd, offsets, args[0], connectorIdToName[args[0]])
+}
+
+func (c *offsetCommand) mapConnectorIdToName(environmentId, kafkaClusterId string) (map[string]string, error) {
+	// NOTE: Do NOT replace this with `V2Client.GetConnectorExpansionById` calls; that function itself calls `V2Client.ListConnectorsWithExpansions`
+	connectors, err := c.V2Client.ListConnectorsWithExpansions(environmentId, kafkaClusterId, "id,info,status")
+	if err != nil {
+		return nil, err
+	}
+
+	connectorIdToName := make(map[string]string)
+	for _, connector := range connectors {
+		connectorIdToName[connector.Id.GetId()] = connector.Info.GetName()
+	}
+
+	return connectorIdToName, nil
+}
+
+func printHumanDescribeOffset(cmd *cobra.Command, offsets connectv1.ConnectV1ConnectorOffsets, id string, name string) error {
+	output.Println(false, "Connector Details")
+	table := output.NewTable(cmd)
+	table.Add(&ConnectDescribeOut{
+		Name: name,
+		Id:   id,
+	})
+	if err := table.Print(); err != nil {
+		return err
+	}
+	output.Println(false, "")
+	output.Println(false, "")
+
+	list := output.NewList(cmd)
+	output.Println(false, "Offset Details")
+	if &offsets != nil && offsets.HasOffsets() {
+		for _, values := range *offsets.Offsets {
+			partition := values["partition"]
+			partitionString, err := json.Marshal(partition)
+			if err != nil {
+				return err
+			}
+			offset := values["offset"]
+			offsetString, err := json.Marshal(offset)
+			if err != nil {
+				return err
+			}
+			list.Add(&offsetsDescribeOut{
+				Partition: string(partitionString),
+				Offset:    string(offsetString),
+			})
+		}
+		if err := list.Print(); err != nil {
+			return err
+		}
+		output.Println(false, "")
+		output.Println(false, "")
+	}
+
+	if &offsets != nil && offsets.HasMetadata() {
+		output.Println(false, "Metadata Details")
+		list = output.NewList(cmd)
+
+		var observedAt string
+		if offsets.Metadata.ObservedAt != nil {
+			observedAt = offsets.Metadata.ObservedAt.String()
+		}
+		list.Add(&metadataDescribeOut{
+			ObservedAt: observedAt,
+		})
+	}
+
+	return list.Print()
+}
+
+func printSerializedDescribeOffsets(cmd *cobra.Command, offsets connectv1.ConnectV1ConnectorOffsets, id string, name string) error {
+
+	var offsetInfo []map[string]interface{}
+	var metadata connectv1.ConnectV1ConnectorOffsetsMetadata
+	if &offsets != nil && offsets.HasMetadata() {
```

Comment:
Taking the memory address of a variable will never not be `nil`

Reply:
Done